components/post_view/post_reaction/index.js

```diff
@@ -3,17 +3,30 @@
 
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
+import {makeGetReactionsForPost} from 'mattermost-redux/selectors/entities/posts';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
 
-import {addReaction} from 'actions/post_actions.jsx';
+import {addReaction, removeReaction} from 'actions/post_actions.jsx';
 
 import PostReaction from './post_reaction';
 
+function mapStateToProps(state, ownProps) {
+    const getReactionsForPost = makeGetReactionsForPost();
+    const currentUserId = getCurrentUserId(state);
+
+    return {
+        reactions: Object.values(getReactionsForPost(state, ownProps.postId) || {}),
```

Comment:
This is going to pass a new array into `PostReaction` every time `mapStateToProps` is called. We should pass in this object into `PostReaction` and have it call `Object.values` on it itself.

Alternatively, instead of making `PostReaction` be in charge of deciding between `addReaction` and `removeReaction`, we could make a new `toggleReaction` action that gets these values from the redux state so that we can avoid passing them into the component in the first place. That could also be used in both of these components.

Reply:
I also thought about doing this but for some reason decided to do a quick fix 🤷  I will do it now :)